cores/esp32/esp32-hal-bt.c

```diff
@@ -20,10 +20,10 @@ bool btInUse(){ return true; }
 
 #include "esp_bt.h"
 
-#ifdef CONFIG_BTDM_CONTROLLER_MODE_BTDM
-#define BT_MODE ESP_BT_MODE_BTDM
-#elif defined(CONFIG_BTDM_CONTROLLER_MODE_BR_EDR_ONLY)
+#ifdef CONFIG_BTDM_CONTROLLER_MODE_BR_EDR_ONLY
 #define BT_MODE ESP_BT_MODE_CLASSIC_BT
+#elif defined(CONFIG_BTDM_CONTROLLER_MODE_BTDM)
+#define BT_MODE ESP_BT_MODE_BTDM
```

Comment:
does switching defines do anything here?

Reply:
My intention is to tell btStart() that I want to use bluetooth-classic only. CONFIG_BTDM_CONTROLLER_MODE_BTDM is defined in a default aduino-esp32 configuration, so it is the first #if that matches and can not be overridden. When we switch the defines BR_EDR_ONLY can be forced with a -DCONFIG_BTDM_CONTROLLER_MODE_BR_EDR_ONLY.

If you prefer an other #define like ARDUINO_BLUETOOTH_EDR_ONLY or a new function for BT-classic initialization please let me know.

Reply:
how about you instead `-DBT_MODE=ESP_BT_MODE_CLASSIC_BT` yourself and only `#ifndef BT_MODE` do the config check.